packages/aws-rfdk/lib/deadline/lib/thinkbox-docker-images.ts

```diff
@@ -0,0 +1,183 @@
+   * @param id The construct ID
+   */
+  public readonly remoteConnectionServer: ContainerImage;
+
+  /**
+   * A {@link DockerImageAsset} that can be used to build Thinkbox's Deadline License Forwarder Docker Recipe into a
+   * container image that can be deployed in CDK.
+   *
+   * @param scope The parent scope
+   * @param id The construct ID
+   */
+  public readonly licenseForwarder: ContainerImage;
+
+  /**
+   * The version of Deadline in the stage directory.
+   */
+  public readonly version?: IVersion;
+
+  constructor(scope: Construct, id: string, props?: ThinkboxDockerImagesProps) {
+    super(scope, id);
+
+    this.version = props?.version;
+
+    const lambdaCode = Code.fromAsset(path.join(__dirname, '..', '..', 'lambdas', 'nodejs'));
+
+    const lambdaFunc = new SingletonFunction(this, 'VersionProviderFunction', {
+      uuid: '08553416-1fc9-4be9-a818-609a31ae1b5b',
+      description: 'Used by the ThinkboxDockerImages construct to obtain the ECR repositories that contain Deadline container images.',
+      code: lambdaCode,
+      runtime: Runtime.NODEJS_12_X,
+      handler: 'ecr-provider.handler',
+      timeout: Duration.seconds(30),
+      logRetention: RetentionDays.ONE_WEEK,
+    });
+
+    const ecrProvider = new CustomResource(this, 'ThinkboxEcrProvider', {
+      serviceToken: lambdaFunc.functionArn,
+      properties: {
+        // create a random string that will force the Lambda to always run on redeploys. Changes to its output will be
+        // propagated to any CloudFormation resource providers that reference the output ARN
+        ForceRun: this.forceRun(),
+      },
+      resourceType: 'Custom::RFDK-EcrProvider',
+    });
+
+    this.node.defaultChild = ecrProvider;
+
+    const versionString = this.versionString(props?.version);
+    const ecrBaseURI = ecrProvider.getAtt('EcrURIPrefix').toString();
+
+    this.remoteConnectionServer = this.globalImageForRecipe(
+      ecrBaseURI,
+      ThinkboxManagedDeadlineDockerRecipes.REMOTE_CONNECTION_SERVER,
+      versionString,
+    );
+    this.licenseForwarder = this.globalImageForRecipe(
+      ecrBaseURI,
+      ThinkboxManagedDeadlineDockerRecipes.LICENSE_FORWARDER,
+      versionString,
+    );
+  }
+
+  private globalImageForRecipe(
+    ecrBaseURI: string,
+    recipe: ThinkboxManagedDeadlineDockerRecipes,
+    version: string | undefined,
+  ) {
+    let registryName = `${ecrBaseURI}${recipe}`;
+    if (version) {
+      registryName += `:${version}`;
+    }
+    return ContainerImage.fromRegistry(
+      registryName,
+    );
+  }
+
+  /**
+   * Returns container images for use with the {@link RenderQueue} construct
+   */
+  public forRenderQueue(): RenderQueueImages {
+    return this;
+  }
+
+  /**
+   * Returns container images for use with the {@link UsageBasedLicensing} construct
+   */
+  public forUsageBasedLicensing(): UsageBasedLicensingImages {
+    return this;
+  }
+
+  private versionString(version?: IVersion) {
```

Comment:
Maybe we could also add a return type here.

Reply:
done